docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,604 @@
 title: Adding search with Algolia
 ---
 
-This is a stub. Help our community expand it.
+Once you've added some content to your site, you'll want to make it easy for your visitors to find what they're looking for. This guide will run you through the process of setting up a custom search experience on any Gatsby site powered by [Algolia](https://www.algolia.com).
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+Two things before we begin:
+
+1. Beyond this guide, you may also want to checkout Algolia's extensive [docs on how to get started in React](https://www.algolia.com/doc/guides/building-search-ui/getting-started/react).
+2. If you're looking to add search to a documentation site, you can let Algolia handle most of the steps outlined below for you by using their excellent [Docsearch](https://community.algolia.com/docsearch) functionality. For other types of sites and more fine-grained control over exactly what data should be indexed, read on.
+
+## Backend
```

Comment:
This is a nitpick, but I'd call this "configuration" instead of "backend". 

```suggestion
## Configuring the Algolia plugin
```

Reply:
Sounds good to me.